backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/internal/BpkSelectImpl.kt

```diff
@@ -0,0 +1,120 @@
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.textfield.internal.BpkTextFieldImpl
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.ArrowDown
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.compose.utils.clickable
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun BpkSelectImpl(
+    options: List<String>,
+    selectedIndex: Int,
+    placeholder: String,
+    modifier: Modifier = Modifier,
+    state: BpkFieldStatus = LocalFieldStatus.current,
+    onSelectionChange: ((selectedIndex: Int) -> Unit)? = null,
+) {
+    var expanded by remember { mutableStateOf(false) }
+    val initialText = if (selectedIndex > -1 && options.size >= selectedIndex) options[selectedIndex] else ""
+    var selectedText by remember { mutableStateOf(initialText) }
+
+    ExposedDropdownMenuBox(
+        expanded = expanded,
+        onExpandedChange = { expanded = !expanded },
+    ) {
+        BpkTextFieldImpl(
+            modifier = modifier.menuAnchor(),
+            value = selectedText,
+            onValueChange = {},
+            readOnly = true,
+            placeholder = placeholder,
+            status = state,
+            trailingIcon = BpkIcon.ArrowDown,
+        )
+        ExposedDropdownMenu(
+            expanded = if (state != BpkFieldStatus.Disabled) expanded else false,
+            modifier = Modifier.background(BpkTheme.colors.surfaceDefault).fillMaxWidth(),
+            onDismissRequest = { expanded = false },
+        ) {
+            options.forEachIndexed { index, option ->
+                val itemBackgroundColor = if (index == selectedIndex) BpkTheme.colors.surfaceHighlight else BpkTheme.colors.surfaceDefault
+                DropdownMenuItem(
+                    modifier = Modifier
+                        .height(BpkSpacing.Lg.times(2))
+                        .background(itemBackgroundColor),
+                    text = { BpkText(text = option, color = BpkTheme.colors.textPrimary, maxLines = 1, overflow = TextOverflow.Ellipsis) },
+                    onClick = {
+                        selectedText = option
+                        expanded = false
+                        if (index != selectedIndex) {
+                            onSelectionChange?.let { it(index) }
+                        }
+                    },
+                    contentPadding = ExposedDropdownMenuDefaults.ItemContentPadding,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+internal fun BpkSelectImpl(
+    placeHolder: String,
+    text: String,
+    modifier: Modifier = Modifier,
+    state: BpkFieldStatus = LocalFieldStatus.current,
+    onClick: (() -> Unit)? = null,
+) {
+    var selectText by remember { mutableStateOf(text) }
+
+    BpkTextFieldImpl(
+        modifier = if (state == BpkFieldStatus.Disabled) modifier else modifier.clickable(bounded = true, role = null) {
```

Comment:
@marianeum I don't know what role should I set here. Does it work as a **button**?

Reply:
yeah button makes sense in this case as we don't know what happens afterwards